app/services/katello/pulp3/repository.rb

```diff
@@ -75,8 +75,39 @@ def content_service
 
       def create_remote
         remote_file_data = api.class.remote_class.new(remote_options)
-        response = api.remotes_api.create(remote_file_data)
-        repo.update!(:remote_href => response.pulp_href)
+        reformat_api_exception do
+          response = api.remotes_api.create(remote_file_data)
+          repo.update!(:remote_href => response.pulp_href)
+        end
+      end
+
+      # When updating a repository, we need to update the remote, but this is
+      # an async task.  If some validation occurs, we won't know about it until
+      # the task runs.  Errors during a repository update task are very difficult to
+      # handle once the task is in its run phase, so this creates a test remote
+      # with a random name in order to validate the remote's configuration
+      def create_test_remote
+        test_remote_options = remote_options
+        test_remote_options[:name] = test_remote_name
+        remote_file_data = api.class.remote_class.new(test_remote_options)
+
+        reformat_api_exception do
+          response = api.remotes_api.create(remote_file_data)
+          #delete is async, but if its not properly deleted, orphan cleanup will take care of it later
+          delete_remote(response.pulp_href)
+        end
+      end
+
+      def test_remote_name
+        "test_remote_#{SecureRandom.uuid}"
+      end
+
+      def reformat_api_exception
+        yield
+      rescue api.class.client_module::ApiError => exception
+        body = JSON.parse(exception.response_body) rescue body
+        body = body.values.join(',') if body.respond_to?(:values)
```

Comment:
Code looks good to me..Except I don't understand this syntax:
`body = JSON.parse(exception.response_body) rescue body`

Reply:
it returns the body if an exception is thrown, for example:
```
1/0 rescue 5
```
returns 5.  In this case i wanted to attempt to parse the response as json, but in case it wasn't just return it as is